test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -955,7 +998,11 @@ public static enum ReadOnlyPathAssert{
         APP_CONTENT(new Builder("--app-content").multiple().create()),
         RESOURCE_DIR(new Builder("--resource-dir").create()),
         MAC_DMG_CONTENT(new Builder("--mac-dmg-content").multiple().create()),
-        RUNTIME_IMAGE(new Builder("--runtime-image").create());
+        RUNTIME_IMAGE(new Builder("--runtime-image").enable(cmd -> {
+            // External runtime image should be R/O unless it is on macOS.
+            // On macOS it will be signed ad-hoc or with real certificate.
+            return !TKit.isOSX();
```

Comment:
Should be
```
return !(TKit.isOSX() && MacHelper.signPredefinedAppImage(cmd));
```

Otherwise, it will be turned off for macOS entirely.

Reply:
Yes, you get the comment right. We do sign runtime image as well (same as app image). We do ad-hoc signing for runtime image if signing is not requested. Ad-hoc signing will modify runtime image. I think you mean case when "--app-image" and "--runtime-image" is specified, then yes we should not modify it, but when we packaging just runtime it will be modify due to signature. I will update test to reflect it.